packages/smooth_app/lib/pages/product/add_basic_details_page.dart

```diff
@@ -0,0 +1,185 @@
+                        if (value == null || value.isEmpty) {
+                          return appLocalizations
+                              .add_basic_details_brand_name_error;
+                        }
+                        return null;
+                      },
+                    ),
+                    SizedBox(height: size.height * 0.02),
+                    SmoothTextFormField(
+                      controller: _weightController,
+                      type: TextFieldTypes.PLAIN_TEXT,
+                      hintText: appLocalizations.weight_gram,
+                      textInputType: TextInputType.number,
+                      validator: (String? value) {
+                        if (value == null || value.isEmpty) {
+                          return appLocalizations
+                              .add_basic_details_weight_gram_error;
+                        }
+                        return null;
+                      },
+                    ),
+                    SizedBox(height: size.height * 0.05),
+                  ],
+                ),
+              ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                _buildButton(
+                  appLocalizations.cancel,
+                  () => Navigator.pop(context),
+                ),
+                _buildButton(appLocalizations.save, () {
+                  if (!_formKey.currentState!.validate()) {
+                    return;
+                  }
+                  saveData();
+                }),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _errormessageAlert(final String message) => showDialog<void>(
+        context: context,
+        builder: (BuildContext context) => SmoothAlertDialog(
+          body: ListTile(
+            leading: const Icon(Icons.error_outline, color: Colors.red),
+            title: Text(message),
+          ),
+          actions: <SmoothActionButton>[
+            SmoothActionButton(
+              text: AppLocalizations.of(context)!.close,
+              onPressed: () => Navigator.pop(context),
+            ),
+          ],
+        ),
+      );
+
+  Future<void> saveData() async {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    widget.product.productName = _productNameController.text;
+    widget.product.quantity = _weightController.text;
+    widget.product.brands = _brandNameController.text;
+    final Status status = await OpenFoodAPIClient.saveProduct(
+      ProductQuery.getUser(),
+      widget.product,
+    );
+    if (status.error != null) {
+      _errormessageAlert(appLocalizations.basic_details_add_error);
+      return;
+    }
+    ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(content: Text(appLocalizations.basic_details_add_success)));
+    Navigator.pop(context);
```

Comment:
Just a personal opinion:
* I would make `_save` return a `String`
* and put all the "what shall we do?" logic at the caller level
* like "I'm the caller, I call `_save`, what's the returned value? `null` is perfect I pop, not null I display the error message"
* it makes more sense= `_save()` just saves, and the caller says what needs to be done depending on the returned value.

Reply:
Suggestion implemented. I also this is a better way to do things.